### ISSUE TITLE
Typeguard corrupt TypeVar 

### ISSUE THREAD

**Issue by tamuhey** [microsoft/pyright#1388]
```python
from typing import TypeVar

T = TypeVar("T")


def id(x: T) -> T:
    if isinstance(x, int):
        return x # Error
    return x # no error
```

Outputs:

```
  8:16 - error: Expression of type "int" cannot be assigned to return type "T@f"
    Type "int" cannot be assigned to type "T@f" (reportGeneralTypeIssues)
```

expected no error

**Comment by erictraut** [comment]
This behavior is "as designed". Mypy gives the same result in this case.
